src/components/search/search-result.js

```diff
@@ -0,0 +1,219 @@
+
+      if (this.shouldFetchListingSchema()) {
+        fetch(`schemas/${this.props.listingType}.json`)
+          .then((response) => response.json())
+          .then((schemaJson) => {
+            this.setState({ listingSchema: schemaJson })
+          })
+      }
+    }
+  }
+
+  // Basic schema validation
+  validateFilterSchema(filterSchemaJson) {
+    filterSchemaJson
+      .items
+      .map(filterGroup => {
+        if (filterGroup.type !== 'filterGroup')
+          throw `Only filterGroup objects are allowed inside items array. Malformed object: ${JSON.stringify(filterGroup)}`
+        else if (!filterGroup.hasOwnProperty("title") || !filterGroup.title.hasOwnProperty("id") || !filterGroup.title.hasOwnProperty("defaultMessage"))
+          throw `Each filterGroup object should have a title object which should consist of "id" and "defaultMessage" properties. Malformed object: ${JSON.stringify(filterGroup)}`
+        else if (!filterGroup.hasOwnProperty("items"))
+          throw `Each filterGroup object should have an "items" member whose value is an array. Malformed object: ${JSON.stringify(filterGroup)}`
+
+        filterGroup
+          .items
+          .map(filter => {
+            if (!filter.hasOwnProperty("type"))
+              throw `Each filter should have a "type" property. Malformed object: ${JSON.stringify(filter)}`
+            else if (!filter.hasOwnProperty("searchParameterName"))
+              throw `Each filter should have a "searchParameterName" property. Malformed object: ${JSON.stringify(filter)}`
+            else if (!/^[a-zA-Z]+$/g.test(filter.searchParameterName))
+              throw `"searchParameterName" property should only consist of english letters. Received: ${filter.searchParameterName}`
+          })
+
+      })
+  }
+
+  formatFiltersToUrl() {
+    //TODO: implement this
+    //document.location.href = `#/search?search_query=${this.state.searchQuery}&listing_type=${this.state.selectedListingType.type}`
+  }
+
+  async searchRequest() {
+    try {
+      this.setState({ searchError: undefined })
+      this.formatFiltersToUrl()
+
+      const searchResponse = await origin.marketplace.search(
+        this.props.query,
+        this.props.listingType,
+        Object.values(this.props.filters).flatMap(arrayOfFilters => arrayOfFilters)
+      )
+
+      if (searchResponse.status !== 200)
+        throw 'Unexpected result received from search engine'
+
+      const json = await searchResponse.json()
+      this.setState({
+        listingIds: json.data.listings.nodes.map(listing => listing.id)
+      })
+
+    } catch (e) {
+      const errorMessage = this.props.intl.formatMessage({
+        id: 'searchResult.canNotReachIndexingServer',
+        defaultMessage: 'We are experiencing some problems. Please try again later'
+      })
+
+      console.error(e)
+      this.props.showAlert(errorMessage)
+      this.setState({ searchError: errorMessage })
+    }
+
+  }
+
+  render() {
+    return (
+      <div>
```

Comment:
Can we use a `Fragment` here instead of an empty `div`?

Reply:
Aaa thanks, wasn't aware `Fragments` are there to avoid using unnecessary divs.